pkg/workers/sharings/share_data.go

```diff
@@ -496,6 +514,46 @@ func sendPatchToRecipient(patch *jsonapi.Document, opts *SendOptions, recipient
 	return err
 }
 
+// getParentDirID returns the id of the parent directory the file should have at
+// the recipient.
+//
+// Two scenarii are possible:
+// 1. There is NO selector: the sharing is based on folders/files. If the file
+//    we are about to send has its id in the `values` declared in the
+//    permissions then it is one of the targets of this sharing. Its
+//    `dirID` must be set to `SharedWithMe`. If not then we don't modify it.
+// 2. There is a selector: the sharing is not based on folders/files. We change
+//    the `dirID` to `SharedWithMe`.
+func getParentDirID(opts *SendOptions, dirID string) (parentID string, err error) {
+	if opts.Selector == "" {
+		if opts.DocID == consts.RootDirID {
```

Comment:
It depends on whether we want to share files that were trashed or not. Right now the behavior is to stop the sharing if the target (i.e. what is shared) is trashed, but if at some point we want to "share" trashed files then yes we could have the same policy here.